File: 3 Loops/3.Nested_Loop3_InvertedTra.cpp
#include<iostream>
using namespace std;



int main() {
    int n=4;
    for (int i=0; i<n; i++) { //Outer
        //Spaces
        for (int j=0;j<i;j++){ // I times
            cout<<" ";
        }
        //Numbers
        for (int j=0; j<n-i; j++) {
            cout<<i+1;
        }
        cout<<endl;
    }
    return 0;
}

File: 3 Loops/3.Nested_Loop3_InvertedTra.py
n = 4
for i in range(n):  # Outer loop
    # Spaces
    spaces = ' ' * i  # Create a string of i spaces

    # Numbers
    numbers = str(i + 1) * (n - i)  # Repeat the number (i + 1), (n - i) times

    print(spaces + numbers)  # Print the spaces and numbers together


# Explanation for Python:
# Outer loop: The outer loop runs from i = 0 to i < n (i.e., 0 to 3 for n = 4).
# Spaces: The spaces string is created by multiplying a space (' ') by i. The number of spaces increases by 1 with each iteration of the outer loop.
# Numbers: The numbers string is created by repeating the string of i + 1 for n - i times.
# Printing: The print(spaces + numbers) statement prints the combined spaces and numbers for each row.